src/nupic/ntypes/Sdr.hpp

```diff
@@ -754,26 +766,99 @@ class SparseDistributedRepresentation : public Serializable
         // Read the dimensions.
         readVector( dimensions_ );
 
-        // Read the data.
-        clear();
-        readVector( flatSparse );
-        flatSparse_valid = true;
-
-        // Consume the end marker.
-        inStream >> marker;
-        NTA_ASSERT( marker == "~SDR" );
-
         // Initialize the SDR.
         // Calculate the SDR's size.
         size_ = 1;
         for(UInt dim : dimensions)
             size_ *= dim;
         // Initialize sparse tuple.
         sparse.assign( dimensions.size(), {} );
+
+        // Read the data.
+        readVector( flatSparse );
+        setFlatSparseInplace();
+
+        // Consume the end marker.
+        inStream >> marker;
+        NTA_ASSERT( marker == "~SDR" );
     };
+
+    void addCallback(SDR_callback_t callback) {
+        callbacks.push_back( callback );
+    }
 };
 
 typedef SparseDistributedRepresentation SDR;
 
+class SDR_Proxy : public SDR
+{
+public:
+    SDR_Proxy(SDR &sdr, const vector<UInt> &dimensions)
+        : SDR( dimensions ) {
+        parent = &sdr;
+        NTA_CHECK( size == parent->size ) << "SDR Proxy must have same size as given SDR.";
+        clear();
+        parent->addCallback( (SDR_callback_t) this );
+    };
+
+    SDR_dense_t& getDense()
+        { return parent->getDense(); }
```

Comment:
...that way you wouldn't have to add these overrides for getters. 

Otherwise, add `override` here and everywhere else applicable, please.